use framework helpers to create namespaces

JarHMJ opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #108298

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

JarHMJ commented:
/cc @s-urbaniak @knabben @caseydavenport

aojea commented:
/ok-to-test
/assign @aojea @liggitt @BenTheElder

leilajal commented:
/triage accepted

Commit:
use framework helpers to create namespaces

JarHMJ commented:
/retest

danwinship commented on test/e2e/network/netpol/network_policy.go:
```diff
@@ -130,10 +130,11 @@ var _ = common.SIGDescribe("Netpol", func() {
 
 		ginkgo.It("should support a 'default-deny-ingress' policy [Feature:NetworkPolicy]", func() {
-			nsX, _, _, k8s := getK8sNamespaces(f)
+			_, _, _, k8s := getK8sNamespaces(f)
```
Every caller to `getK8sNamespaces` now ignores the first three return values. If the method is going to stay the way it is, you should just remove those return values. But it would be better to fix things so that it knows the real namespace name again.

danwinship commented on test/e2e/network/scale/localrun/ingress_scale.go:
```diff
@@ -135,7 +135,6 @@ func main() {
 			Name: testNamespace,
 			Labels: map[string]string{
-				// TODO(https://github.com/kubernetes/kubernetes/issues/108298): route namespace creation via framework.Framework.CreateNamespace in 1.24
 				admissionapi.EnforceLevelLabel: string(admissionapi.LevelPrivileged),
```
this removes the TODO without actually changing it to use the helper function...

danwinship reviewed (changes requested):
networking bits (at least) need further refactoring

antoninbas commented:
@JarHMJ are you planning on addressing comments from @danwinship and updating this PR? I am asking because I am running into an issue which is addressed by this PR (see https://github.com/kubernetes/kubernetes/issues/108298#issuecomment-1197236231). If you don't have cycles to work on this item any more, please let me know and I could pick it up.

JarHMJ commented:
> @JarHMJ are you planning on addressing comments from @danwinship and updating this PR? I am asking because I am running into an issue which is addressed by this PR (see [#108298 (comment)](https://github.com/kubernetes/kubernetes/issues/108298#issuecomment-1197236231)). If you don't have cycles to work on this item any more, please let me know and I could pick it up.

yes, I have no time to update this pr recently and you are welcome to pick it up. @antoninbas

antoninbas commented:
Superseded by https://github.com/kubernetes/kubernetes/pull/111789 for Netpol tests. @danwinship would appreciate if you could take a look.

antoninbas commented:
@JarHMJ I think you can close this PR now, mine has been merged

dims commented:
If you still need this PR then please rebase, if not, please close the PR

aojea commented:
/close

it seems this PR already fixed the mentioned issue
https://github.com/kubernetes/kubernetes/pull/112019
